code/game/machinery/doors/airlock.dm

```diff
@@ -1083,10 +1088,41 @@ About the new airlock wires panel:
 	use_power(360)	//360 W seems much more appropriate for an actuator moving an industrial door capable of crushing people
 
 	//if the door is unpowered then it doesn't make sense to hear the woosh of a pneumatic actuator
-	if(arePowerSystemsOn())
-		playsound(src, open_sound_powered, 50, 1, volume_channel = VOLUME_CHANNEL_DOORS)
-	else
-		playsound(src, open_sound_unpowered, 75, 1, volume_channel = VOLUME_CHANNEL_DOORS)
+	for(var/P in player_list)

```

Comment:
I feel like it may be more efficient to just move all of this into a `playsound_if_pref_enabled()` proc that just takes ~~the pref-on and pref-off sounds and the path of the pref to check~~ matching lists of sounds and preferences to check, in order of priority, in addition to all of playsound()'s arguments. I think a list of toggles is more extensible than trying to put the two preferences into a single numeric pref that acts as a finite-state-machine.

Certainly better than just copypasting it into doorcode and making it kinda snowflakey